src/eureka/S3_data_reduction/plots_s3.py

```diff
@@ -565,3 +568,307 @@ def median_frame(data, meta):
+                    extent=[xmin, xmax, ymin, ymax])
+    plt.scatter(centroid_x, centroid_y, marker='x', s=25, c='r', label='centroid')
+    plt.title('Full Frame')
+    plt.ylabel('y pixels')
+    plt.xlabel('x pixels')
+
+    circle1 = plt.Circle((centroid_x, centroid_y), meta.photap, color='r',
+                         fill=False, lw=3, alpha=0.7, label='target aperture')
+    circle2 = plt.Circle((centroid_x, centroid_y), meta.skyin, color='w',
+                         fill=False, lw=4, alpha=0.8, label='sky aperture')
+    circle3 = plt.Circle((centroid_x, centroid_y), meta.skyout, color='w',
+                         fill=False, lw=4, alpha=0.8)
+    plt.gca().add_patch(circle1)
+    plt.gca().add_patch(circle2)
+    plt.gca().add_patch(circle3)
+
+    add_colorbar(im, label='Flux (electrons)')
+    xlim_min = max(0, centroid_x - meta.skyout - 10)
+    xlim_max = min(centroid_x + meta.skyout + 10, flux.shape[1])
+    ylim_min = max(0, centroid_y - meta.skyout - 10)
+    ylim_max = min(centroid_y + meta.skyout + 10, flux.shape[0])
+
+    plt.xlim(xlim_min, xlim_max)
+    plt.ylim(ylim_min, ylim_max)
+    plt.xlabel('x pixels')
+    plt.ylabel('y pixels')
+
+    plt.legend()
+    plt.tight_layout()
+    file_number = str(m).zfill(int(np.floor(np.log10(meta.num_data_files))+1))
+    int_number = str(i).zfill(int(np.floor(np.log10(meta.n_int))+1))
+
+    fname = (f'figs{os.sep}fig3504_file{file_number}_int{int_number}_2D_Frame_Zoom'
+             + figure_filetype)
+    plt.savefig(meta.outputdir + fname, dpi=250, tight_layout=True)
+    if not meta.hide_plots:
+        plt.pause(0.2)
+
+
+def phot_2d_frame_oof(meta, m, i, data, flux_w_oof):
+    """
+    Plots the 2D frame together with the centroid position and apertures.
+    """
+    fig, ax = plt.subplots(2, 1, figsize=(8.2, 4.2))
+
+    ax[0].imshow(flux_w_oof, origin='lower', norm=LogNorm(vmin=0.1, vmax=40), cmap='viridis')
+    ax[0].set_title('Before 1/f correction')
+    ax[0].set_ylabel('y pixels')
+
+    flux = data.flux.values[i]
+    im1 = ax[1].imshow(flux, origin='lower', norm=LogNorm(vmin=0.1, vmax=40), cmap='viridis')
+    ax[1].set_title('After 1/f correction')
+    ax[1].set_xlabel('x pixels')
+    ax[1].set_ylabel('y pixels')
+
+    plt.subplots_adjust(hspace=0.3)
+
+    cbar = fig.colorbar(im1, ax=ax)
+    cbar.ax.set_ylabel('Flux (electrons)')
+    file_number = str(m).zfill(int(np.floor(np.log10(meta.num_data_files))+1))
+    int_number = str(i).zfill(int(np.floor(np.log10(meta.n_int))+1))
+    fig.suptitle((f'Segment {file_number}, Integration {int_number}'), y=0.99)
+
+    fname = (f'figs{os.sep}fig3307_file{file_number}_int{int_number}_2D_Frame_OOF' +
+             figure_filetype)
+    plt.savefig(meta.outputdir + fname, dpi=250, tight_layout=True)
+    if not meta.hide_plots:
+        plt.pause(0.2)
+
+
+def phot_2d_frame_diff(meta, data):
+    """
+    Plots the 2D frame together with the centroid position and apertures.
```

Comment:
Please avoid duplicated docstrings like this one - what does this specific function make?